src/Razor/src/Microsoft.AspNetCore.Razor.VSCode.BlazorWasmCompanionExtension/src/LocalDebugProxyManager.ts

```diff
@@ -0,0 +1,65 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+const fetch = require('node-fetch');
+import * as stream from 'stream';
+import * as extract from 'extract-zip';
+import * as net from 'net';
+import * as os from 'os';
+import * as fs from 'fs';
+import * as util from 'util';
+
+export class LocalDebugProxyManager {
+    private readonly localPath = `${os.homedir()}/.blazor-wasm-debug-proxy-local`;
+    private readonly nugetUrl = 'https://api.nuget.org/v3-flatcontainer';
+    private readonly packageName = 'Microsoft.AspNetCore.Components.WebAssembly.DevServer';
+    private readonly finished = util.promisify(stream.finished);
+
+    constructor() {
+        if (!fs.existsSync(this.localPath)) {
+            fs.mkdirSync(this.localPath);
```

Comment:
Good call! Will opt for an `accessSync` check here which validates both existence and permissions.

Reply:
We'll also want to ensure we explode in a meaningful manner for users so they and us know why things are broke

Reply:
I think leveraging the `storagePath` context mentioned above will help us avoid this problem since VS Code provides guarantees about access/availability of that path.